### ISSUE TITLE
Handle NULL return value for spg_get_dataset()

### ISSUE THREAD

**Issue by kbarros** [singularitti/Spglib.jl#76]
The following will segfault:

```
lat_vecs = [4.05 -2.025 0.0; 0.0 3.507402885326976 0.0; 0.0 0.0 6.76]
positions = [0.6666 0.3333 0.75; 0.6667 0.3334 0.75]'
symprec = 1e-3
cell = Spglib.Cell(lat_vecs, positions, [1, 1])
d = Spglib.get_dataset(cell, symprec) # crashes
```

The problem is that the unit cell is invalid for `symprec=1e-3`, and `spg_get_dataset()` returns `NULL`. However, the Spglib.jl bindings do not handle the `NULL` case here: https://github.com/singularitti/Spglib.jl/blob/master/src/symmetry.jl#L259

In case of failure, it seems most appropriate for `Spglib.get_dataset` to return `nothing`.

**Comment by kbarros** [comment]
It seems that `spg_get_dataset` returning `NULL` instead of space group 0 is a new behavior as of spglib 1.8.1, per http://spglib.github.io/spglib/api.html#spg-get-dataset-and-spg-get-dataset-with-hall-number

The documentation above also states: "allocated memory space must be freed by `calling spg_free_dataset`." Do the Julia bindings attempt to free this memory?

**Comment by hsugawa8651** [comment]
* Current `Spglib.jl` imports [spglib_jll](https://github.com/JuliaBinaryWrappers/spglib_jll.jl), birary wrapper of `spglib 1.16.0` . According to its C-source, `get_dataset` function returns `NULL` if failed. See at 
https://github.com/spglib/spglib/blob/e8118d854a4c11dbaa8d7b2c55d4a1e74ddcaaf7/src/spglib.c#L1265
(in despite of the documentation' notice  `Changed in version 1.8.1`)

* Python interface of `spglib` comes with wrapper of  `get_dataset` function, which constructs Python's `dataset` object and immediately deallocates C's `dataset` struct via  calling `spg_free_dataset`.  See `py_get_dataset` at 
https://github.com/spglib/spglib/blob/f8e6d5be6703466fa1444500a08d3fa623a0091d/python/_spglib.c#L214

**Comment by singularitti** [comment]
Hey guys, I am no expert in C. So I cannot decide anything. If you would help, I could just merge your PRs.

**Comment by singularitti** [comment]
Closed by #77
